ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/peervalidation/RequiredBlocksPeerValidatorTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ *
+ *  * Copyright ConsenSys AG.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ *  * the License. You may obtain a copy of the License at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ *  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *  * specific language governing permissions and limitations under the License.
+ *  *
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+package org.hyperledger.besu.ethereum.eth.peervalidation;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.hyperledger.besu.ethereum.core.Block;
+import org.hyperledger.besu.ethereum.core.BlockDataGenerator;
+import org.hyperledger.besu.ethereum.core.BlockDataGenerator.BlockOptions;
+import org.hyperledger.besu.ethereum.core.Hash;
+import org.hyperledger.besu.ethereum.eth.manager.EthProtocolManager;
+import org.hyperledger.besu.ethereum.eth.manager.EthProtocolManagerTestUtil;
+import org.hyperledger.besu.ethereum.eth.manager.RespondingEthPeer;
+import org.hyperledger.besu.ethereum.mainnet.MainnetProtocolSchedule;
+import org.hyperledger.besu.metrics.noop.NoOpMetricsSystem;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.junit.Test;
+
+public class RequiredBlocksPeerValidatorTest extends AbstractPeerBlockValidatorTest {
+
+  @Override
+  AbstractPeerBlockValidator getSampleValidator(final long blockNumber) {
+    return new RequiredBlocksPeerValidator(
+        MainnetProtocolSchedule.create(), new NoOpMetricsSystem(), blockNumber, Hash.ZERO, 0);
+  }
+
+  @Test
+  public void validatePeer_responsivePeerWithRequiredBlock() {
```

Comment:
My concern is it would make it difficult to read what is actually being tested.  By factoring too much out a reader would have to go to multiple places and decode what is happening.

Reply:
Yeah, I agree it can get confusing.  But if the basic test cases are the same, it does feel kind of silly to just copy / paste them modifying only a few lines. And I think we could write it in a way that's still pretty readable.  We can always flatten them out later if it gets to be too much.  But I'll leave it up to you 🤷‍♀